### PR TITLE
chore: convert to monorepo

### PR DESCRIPTION
**Pull request by achingbrain**
Converts this repository to a monorepo - in the initial pass the only monorepo package is libp2p itself.

Other packages can be added in follow up PRs.

Refs #824

### CODE REVIEW

File: ISSUE_TEMPLATE.md
Comment (maschad) [inline ISSUE_TEMPLATE.md:1]: Why has this been deleted?

File: MIGRATION_TEMPLATE.md
Comment (maschad) [inline MIGRATION_TEMPLATE.md:1]: Why has this been deleted?

File: OKR.md
Comment (maschad) [inline OKR.md:1]: Tbh this should be removed imo, @p-shahi do you agree?

File: .release-please-manifest.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "packages/libp2p":"0.45.4"
```
Comment (maschad) [inline .release-please-manifest.json:2]: Should this be `0.45.5` ?

**Review by maschad — CHANGES_REQUESTED** [review]
Thanks for the work on this! Will hopefully improve both DX and release management 🥳  I left some comments.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -159,18 +159,34 @@ jobs:
...
+    ]
     if: github.event_name == 'push' && github.ref == 'refs/heads/master'
     steps:
       - uses: GoogleCloudPlatform/release-please-action@v2
         id: release
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+          command: manifest
           release-type: node
-          # breaking changes should rev the minor (remove after v1)
```
[the remark this answers is not in the document]
Comment (achingbrain) [reply .github/workflows/main.yml:170]: This was copied from the js-ipfs workflow which uses release please to do gated monorepo releases.

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -159,18 +159,34 @@ jobs:
...
-          bump-minor-pre-major: true
-          # features should rev the patch (remove after v1)
-          bump-patch-for-minor-pre-major: true
+          manifest-file: .release-please-manifest.json
+          config-file: .release-please.json
+          changelog-types: |
```
[the remark this answers is not in the document]
Comment (achingbrain) [reply .github/workflows/main.yml:182]: Yes, this was copied from the js-ipfs workflow which uses release please to do gated monorepo releases.

File: ISSUE_TEMPLATE.md
Comment (achingbrain) [reply ISSUE_TEMPLATE.md:1]: Because the issue template is in the `.github` folder.

File: MIGRATION_TEMPLATE.md
Comment (achingbrain) [reply MIGRATION_TEMPLATE.md:1]: It seemed redundant.

File: OKR.md
Comment (achingbrain) [reply OKR.md:1]: It's very out of date and has been superseded by the roadmap doc.

File: .release-please-manifest.json (hunk 1 above)
Comment (achingbrain) [reply .release-please-manifest.json:2]: Yeah, it's a pain keeping this PR up to date.

File: .github/workflows/main.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (maschad) [reply .github/workflows/main.yml:182]: Gotcha, and release please looks for [Conventional Commit messages](https://www.conventionalcommits.org/) to automate our releases.  I personally prefer this style of conventional commits but I was just making a suggestion for a discussion before going ahead with it.

File: MIGRATION_TEMPLATE.md
Comment (maschad) [reply MIGRATION_TEMPLATE.md:1]: I think it's useful to explain how we derive our [migrations format](https://github.com/libp2p/js-libp2p/tree/master/doc/migrations) 🤷🏾

File: .github/workflows/main.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (maschad) [reply .github/workflows/main.yml:170]: It doesn't seemed to be used in our workflow anywhere either

**Review by maschad — APPROVED** [review]
I have a [minor comment](https://github.com/libp2p/js-libp2p/pull/1792#discussion_r1223303940) regarding the retention of the migration template but that shouldn't prevent this from being merged.

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -159,18 +159,34 @@ jobs:
...
-    needs: [test-node, test-chrome, test-chrome-webworker, test-firefox, test-firefox-webworker, test-electron-main, test-electron-renderer, test-interop]
+    needs: [
+      test-node,
+      test-chrome,
+      test-chrome-webworker,
+      test-firefox,
+      test-firefox-webworker,
+      test-electron-main,
+      test-electron-renderer,
+      test-interop
```
Comment (wemeetagain) [inline .github/workflows/main.yml:170]: test-examples?

Comment (achingbrain) [reply .github/workflows/main.yml:170]: Examples are tested in a separate workflow file, I don't think you can depend on jobs defined in other files yet.

File: .github/workflows/main.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (achingbrain) [reply .github/workflows/main.yml:182]: There's not much to discuss - the commit style is the same as it was before, this is just config necessary to get release please to perform the gated releases as before.

File: MIGRATION_TEMPLATE.md
Comment (achingbrain) [reply MIGRATION_TEMPLATE.md:1]: Ok, I've moved the template to the migrations dir.

**Comment by achingbrain** [comment]
Merging because keeping this up to date is very painful.

Multidim interop is failing - https://github.com/libp2p/js-libp2p/issues/1809

**Pull request merged**